packages/in-app-notification-dispatcher/src/handlers/handleNewEServiceVersionPublished.ts

```diff
@@ -0,0 +1,54 @@
+import { EServiceV2, fromEServiceV2 } from "pagopa-interop-models";
+import { Logger } from "pagopa-interop-commons";
+import { ReadModelServiceSQL } from "../services/readModelServiceSQL.js";
+import { InAppNotificationServiceSQL } from "../services/inAppNotificationServiceSQL.js";
+import { config } from "../config/config.js";
+import { inAppTemplates } from "../templates/inAppTemplates.js";
+import {
+  retrieveLatestPublishedDescriptor,
+  retrieveTenant,
+} from "./handlerCommons.js";
+
+export async function handleNewEServiceVersionPublished(
+  eserviceV2Msg: EServiceV2,
+  logger: Logger,
+  readModelService: ReadModelServiceSQL,
+  inAppNotificationService: InAppNotificationServiceSQL
+): Promise<void> {
+  logger.info(
+    `Sending in-app notification for new descriptor published eservice ${eserviceV2Msg.id}`
+  );
+
+  const eservice = fromEServiceV2(eserviceV2Msg);
+  const descriptor = retrieveLatestPublishedDescriptor(eservice);
+
+  const [agreements] = await Promise.all([
+    readModelService.getAgreementsByEserviceId(eservice.id),
+  ]);
+
+  if (agreements && agreements.length > 0) {
+    const consumers = await Promise.all(
+      agreements.map((consumer) =>
+        retrieveTenant(consumer.consumerId, readModelService)
+      )
+    );
+    const userNotificationConfigs =
+      await readModelService.getUserNotificationConfigsByTenantIds(
+        consumers.map((consumer) => consumer.id),
+        "newEServiceVersionPublished"
+      );
+
+    const body = inAppTemplates.newEServiceVersionPublished(eservice.name);
+
+    const notifications = userNotificationConfigs.map(
+      ({ userId, tenantId }) => ({
+        userId,
+        tenantId,
+        body,
+        deepLink: `https://${config.interopFeBaseUrl}/ui/it/fruizione/catalogo-e-service/${eservice.id}/${descriptor.id}`,
+      })
+    );
+
+    await inAppNotificationService.insertNotifications(notifications);
```

Comment:
IMO it would be nice to decouple the business logic and I/O a bit more in the handler, to simplify the tests

one way would be to return a list of notifications and move the actual insertion into `index.ts` after the handler call (though it would be repeated there for every handler, unless we refactor that structure a bit to call it in just one place)

this would allow us to test the handler without inspecting the DB but just checking the output array

to go further, we'd have to decouple DB reads as well, which would allow to test some of the logic without the DB at all (speeding up the tests): but probably that's not worth it, since the logic of DB reads is itself interesting to test